File: app/src/main/java/net/seismos/android/seismos/ui/global/DashActivity.java
package net.seismos.android.seismos.ui.global;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import com.google.android.gms.maps.model.LatLng;
import com.google.android.material.bottomnavigation.BottomNavigationView;

import net.seismos.android.seismos.R;
import net.seismos.android.seismos.ui.home.HomeFragment;
import net.seismos.android.seismos.ui.map.MapFragment;
import net.seismos.android.seismos.ui.profile.ProfileFragment;
import net.seismos.android.seismos.ui.seismos.SeismosFragment;
import net.seismos.android.seismos.ui.store.StoreFragment;

public class DashActivity extends AppCompatActivity implements HomeFragment.OnEqGlobeSelectedListener {

    final String HOME_FRAGMENT = "HOME_FRAGMENT";
    final String MAP_FRAGMENT = "MAP_FRAGMENT";
    final String SEISMOS_FRAGMENT = "SEISMOS_FRAGMENT";
    final String STORE_FRAGMENT = "STORE_FRAGMENT";
    final String PROFILE_FRAGMENT = "PROFILE_FRAGMENT";

    HomeFragment homeFragment;
    MapFragment mapFragment;
    SeismosFragment seismosFragment;
    StoreFragment storeFragment;
    ProfileFragment profileFragment;

    private Fragment active;
    private BottomNavigationView nav;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_dash);


        if (savedInstanceState==null) {

            homeFragment = new HomeFragment();
            mapFragment = new MapFragment();
            seismosFragment = new SeismosFragment();
            storeFragment = new StoreFragment();
            profileFragment = new ProfileFragment();


            getSupportFragmentManager().beginTransaction()
                    .add(R.id.host_container,homeFragment, HOME_FRAGMENT)
                    .hide(homeFragment)
                    .add(R.id.host_container, mapFragment, MAP_FRAGMENT)
                    .hide(mapFragment)
                    .add(R.id.host_container, seismosFragment, SEISMOS_FRAGMENT)
                    .hide(seismosFragment)
                    .add(R.id.host_container, storeFragment, STORE_FRAGMENT)
                    .hide(storeFragment)
                    .add(R.id.host_container, profileFragment, PROFILE_FRAGMENT)
                    .hide(profileFragment)
                    .show(homeFragment)
                    .commit();

            active = homeFragment;
        }




        nav = findViewById(R.id.bottom_nav);
        setupNavMenu(nav);
    }




    private void setupNavMenu(BottomNavigationView nav) {
        nav.setOnNavigationItemSelectedListener(menuItem -> {
            switch (menuItem.getItemId()) {
                case R.id.homeTitle:
                    getSupportFragmentManager().beginTransaction()
                            .hide(active)
                            .show(homeFragment)
                            .commit();
                    active = homeFragment;
                    break;
                case R.id.map:
                    getSupportFragmentManager().beginTransaction()
                            .hide(active)
                            .show(mapFragment)
                            .commit();
                    active = mapFragment;
                    break;
                case R.id.seismos:
                    getSupportFragmentManager().beginTransaction()
                            .hide(active)
                            .show(seismosFragment)
                            .commit();
                    active = seismosFragment;
                    break;
                case R.id.store:
                    getSupportFragmentManager().beginTransaction()
                            .hide(active)
                            .show(storeFragment)
                            .commit();
                    active = storeFragment;
                    break;
                case R.id.profile:
                    getSupportFragmentManager().beginTransaction()
                            .hide(active)
                            .show(profileFragment)
                            .commit();
                    active = profileFragment;
                    break;
            }
            return true;
        });
    }


    @Override
    public void openMapToLatLng(LatLng latLng) {
        nav.setSelectedItemId(R.id.map);
        mapFragment.navigateToLatLng(latLng);
    }
}

File: app/src/main/java/net/seismos/android/seismos/ui/global/DashActivity.kt
/*
 * Copyright 2019, The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package net.seismos.android.seismos.ui.global

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.os.bundleOf
import androidx.core.view.get
import androidx.lifecycle.LiveData
import androidx.navigation.NavController
import androidx.navigation.Navigation
import com.google.android.gms.maps.model.LatLng
import com.google.android.material.bottomnavigation.BottomNavigationView
import net.seismos.android.seismos.R

/**
 * An activity that inflates a layout that has a [BottomNavigationView].
 */
class DashActivity : AppCompatActivity() {

    private var currentNavController: LiveData<NavController>? = null
    private var coordinates : DoubleArray? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_dash)
        if (savedInstanceState == null) {
            setupBottomNavigationBar()
        } // Else, need to wait for onRestoreInstanceState
    }

    override fun onRestoreInstanceState(savedInstanceState: Bundle?) {
        super.onRestoreInstanceState(savedInstanceState)
        // Now that BottomNavigationBar has restored its instance state
        // and its selectedItemId, we can proceed with setting up the
        // BottomNavigationBar with Navigation
        setupBottomNavigationBar()
    }

    /**
     * Called on first creation and when restoring state.
     */
    private fun setupBottomNavigationBar() {
        val bottomNavigationView = findViewById<BottomNavigationView>(R.id.bottom_nav)

        val navGraphIds = listOf(R.navigation.home, R.navigation.map, R.navigation.seismos,
                                    R.navigation.store, R.navigation.profile);

        // Setup the bottom navigation view with a list of navigation graphs
        val controller = bottomNavigationView.setupWithNavController(
            navGraphIds = navGraphIds,
            fragmentManager = supportFragmentManager,
            containerId = R.id.nav_host_container,
            intent = intent
        )

//        // Whenever the selected controller changes, setup the action bar.
//        controller.observe(this, Observer { navController ->
//            setupActionBarWithNavController(navController)
//        })
        currentNavController = controller
    }

    override fun onSupportNavigateUp(): Boolean {
        return currentNavController?.value?.navigateUp() ?: false
    }

    fun openMapToLatLng(latLng : LatLng) {
        val lat = latLng.latitude
        val lng = latLng.longitude
        coordinates = doubleArrayOf(lat, lng)
        val bottomNavigationView = findViewById<BottomNavigationView>(R.id.bottom_nav)


        bottomNavigationView.setSelectedItemId(R.id.map)


//        val bundle = bundleOf("lat" to lat)
//        currentNavController?.value?.navigate(R.id.mapFragment, bundle)
    }

     fun getCoord() : DoubleArray? {
         return coordinates
    }

    fun resetCoord() {
        coordinates = null
    }
}
